integration/map_test.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"testing"
+
+	"github.com/google/trillian"
+	"github.com/google/trillian/merkle"
+	_ "github.com/google/trillian/merkle/coniks" // Register
+	"github.com/google/trillian/merkle/hashers"
+	_ "github.com/google/trillian/merkle/maphasher" // Register
+	"github.com/google/trillian/testonly"
+	"github.com/google/trillian/testonly/integration"
+)
+
+// createHashKV returns a []*trillian.MapLeaf formed by the mapping of index, value ...
+// createHashKV panics if len(iv) is odd. Duplicate i/v pairs get over written.
+func createMapLeaves(iv ...[]byte) []*trillian.MapLeaf {
+	if len(iv)%2 != 0 {
+		panic(fmt.Sprintf("integration: createMapLeaves got odd number of iv pairs: %v", len(iv)))
+	}
+	r := []*trillian.MapLeaf{}
+	var index []byte
+	for i, b := range iv {
+		if i%2 == 0 {
+			index = b
+			continue
+		}
+		r = append(r, &trillian.MapLeaf{
+			Index:     index,
+			LeafValue: b,
+		})
+	}
+	return r
+}
+
+func TestInclusionWithEnv(t *testing.T) {
+	ctx := context.Background()
+	env, err := integration.NewMapEnv(ctx, "TestInclusionWithEnv")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer env.Close()
+	for _, tc := range []struct {
+		trillian.HashStrategy
+		index []byte
+		value []byte
+	}{
+		{trillian.HashStrategy_TEST_MAP_HASHER, testonly.TransparentHash("A"), []byte("A")},
+	} {
+		tree, err := env.CreateMap(tc.HashStrategy)
+		if err != nil {
+			t.Errorf("CreateMap(): %v", err)
+			continue
+		}
+		mapID := tree.TreeId
+		hasher, err := hashers.NewMapHasher(tree.HashStrategy)
+		if err != nil {
+			t.Errorf("NewMapHasher(): %v", err)
+			continue
+		}
+		client := trillian.NewTrillianMapClient(env.ClientConn)
```

Comment:
I think handing back both Admin and MapClient is worth it and will pay itself back rather quickly. That's what we're really after when getting a MapEnv, isn't it? The other fields don't matter nearly as much (not even ClientConn, assuming we hand back the specific clients needed).